Forge.Forms/src/Forge.Forms/DynamicExpressions/MultiValueConverters/DivideMultiConverter.cs

```diff
@@ -0,0 +1,188 @@
+                        return f1 / i2;
+                    }
+                    else if (values[1] is short s2)
+                    {
+                        return f1 / s2;
+                    }
+                }
+                else if (values[0] is long l1)
+                {
+                    if (values[1] is decimal dc2)
+                    {
+                        return l1 / (double)dc2;
+                    }
+                    else if (values[1] is double d2)
+                    {
+                        return l1 / d2;
+                    }
+                    else if (values[1] is float f2)
+                    {
+                        return l1 / f2;
+                    }
+                    else if (values[1] is long l2)
+                    {
+                        return l1 / l2;
+                    }
+                    else if (values[1] is int i2)
+                    {
+                        return l1 / i2;
+                    }
+                    else if (values[1] is short s2)
+                    {
+                        return l1 / s2;
+                    }
+                }
+                else if (values[0] is int i1)
+                {
+                    if (values[1] is decimal dc2)
+                    {
+                        return i1 / (double)dc2;
+                    }
+                    else if (values[1] is double d2)
+                    {
+                        return i1 / d2;
+                    }
+                    else if (values[1] is float f2)
+                    {
+                        return i1 / f2;
+                    }
+                    else if (values[1] is long l2)
+                    {
+                        return i1 / l2;
+                    }
+                    else if (values[1] is int i2)
+                    {
+                        return i1 / i2;
+                    }
+                    else if (values[1] is short s2)
+                    {
+                        return i1 / s2;
+                    }
+                }
+                else if (values[0] is short s1)
+                {
+                    if (values[1] is decimal dc2)
+                    {
+                        return s1 / (double)dc2;
+                    }
+                    else if (values[1] is double d2)
+                    {
+                        return s1 / d2;
+                    }
+                    else if (values[1] is float f2)
+                    {
+                        return s1 / f2;
+                    }
+                    else if (values[1] is long l2)
+                    {
+                        return s1 / l2;
+                    }
+                    else if (values[1] is int i2)
+                    {
+                        return s1 / i2;
+                    }
+                    else if (values[1] is short s2)
+                    {
+                        return s1 / s2;
+                    }
```

Comment:
I think all of this can be shortened to one or two lines by utilizing the `IConvertible` interface.
It would look something like this: `Convert.ToDouble(s1) / Convert.ToDouble(s2)`.
Integer conversion would probably need to be handled separately, otherwise we'd end up with decimal numbers after dividing, which may or may not be desired.

If the above approach won't work then C# 7 pattern matching would help in shortening this code.